lib/wallet/error.rs

```diff
@@ -144,6 +190,249 @@ pub enum WalletSync {
+    #[error("failed to decrypt mnemonic")]
+    Decrypt(#[from] aes_gcm::Error),
+    #[error(transparent)]
+    ParseMnemonic(#[from] ParseMnemonic),
+    #[error(transparent)]
+    StretchPassword(#[from] StretchPassword),
+    #[error("failed to convert mnemonic to string")]
+    Utf8(#[from] std::string::FromUtf8Error),
+}
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum UnlockExistingWallet {
+    #[error(transparent)]
+    InitWalletFromMnemonic(#[from] InitWalletFromMnemonic),
+    #[error("wallet is not encrypted")]
+    NotEncrypted,
+    #[error(transparent)]
+    ReadDbMnemonic(#[from] ReadDbMnemonic),
+    #[error(transparent)]
+    WalletInitialization(#[from] WalletInitialization),
+}
+
+impl ToStatus for UnlockExistingWallet {
+    fn builder(&self) -> StatusBuilder {
+        match self {
+            Self::InitWalletFromMnemonic(err) => err.builder(),
+            Self::NotEncrypted => StatusBuilder::new(self),
+            Self::ReadDbMnemonic(err) => err.builder(),
+            Self::WalletInitialization(err) => err.builder(),
+        }
+    }
+}
+
+#[derive(Debug, Diagnostic, Error)]
+#[error("unsupported network (`{0}`)")]
+pub struct UnsupportedNetwork(pub bitcoin::Network);
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum InitEsploraClient {
+    #[error("failed to build esplora client")]
+    BuildEsploraClient(#[source] esplora_client::Error),
+    #[error("failed to get esplora height")]
+    EsploraClientHeight(#[source] esplora_client::Error),
+    #[error(transparent)]
+    ParseUrl(#[from] url::ParseError),
+    #[error(transparent)]
+    UnsupportedNetwork(#[from] UnsupportedNetwork),
+}
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum InitElectrumClient {
+    #[error(
+        "electrum server (`{}`) is not on the same chain as the wallet (`{}`)",
+        .electrum_block_hash,
+        .wallet_chain_hash
+    )]
+    ChainMismatch {
+        electrum_block_hash: bitcoin::BlockHash,
+        wallet_chain_hash: bitcoin::blockdata::constants::ChainHash,
+    },
+    #[error("failed to create electrum client")]
+    CreateElectrumClient(#[source] bdk_electrum::electrum_client::Error),
+    #[error("failed to get initial block header")]
+    GetInitialBlockHeader(#[source] bdk_electrum::electrum_client::Error),
+    #[error(transparent)]
+    UnsupportedNetwork(#[from] UnsupportedNetwork),
+}
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum InitDbConnection {
+    #[error(transparent)]
+    Migration(#[from] rusqlite_migration::Error),
+    #[error(transparent)]
+    Rusqlite(#[from] rusqlite::Error),
+}
+
+type Persistence = <crate::wallet::Persistence as bdk_wallet::AsyncWalletPersister>::Error;
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum InitWalletFromMnemonic {
+    #[error("failed to create wallet")]
+    CreateWallet(#[from] bdk_wallet::CreateWithPersistError<Persistence>),
+    #[diagnostic(transparent)]
+    #[error("wallet data mismatch! Wipe your data directory and try again.")]
```

Comment:
might be better if we have the error message as `wallet data mismatch (wipe your data directory and try again)`

Capital letters and dots compose poorly with nested errors